04_numeronym/shahriys/main.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+)
+
+var outnum io.Writer = os.Stdout
+
+func numeronyms(vals ...string) []string {
+	var returnval = make([]string, 0, len(vals))
+	for _, val := range vals {
+		switch len(val) {
```

Comment:
Interesting use of a `switch` as opposed to an `if` block. Perfectly fine though
You can assign a variable in the `switch` statement and reuse it without having to recompute it
```
switch length := len(vals) {
  case 0, 1, 2, 3:
    // code
  default: 
    l := length - 2
}
```